src/select/Select.ts

```diff
@@ -0,0 +1,411 @@
+	selected?: boolean;
+}
+
+/**
+ * @type SelectProperties
+ *
+ * Properties that can be set on a Select component
+ *
+ * @property describedBy	ID of an element that provides more descriptive text
+ * @property disabled			Prevents the user from interacting with the form field
+ * @property formId				ID of a form element associated with the form field
+ * @property invalid			Indicates the value entered in the form field is invalid
+ * @property label				Label settings for form label text, position, and visibility
+ * @property multiple			Whether the widget supports multiple selection
+ * @property name					The form widget's name
+ * @property options			Array of data for the select options' value, text content, and state
+ * @property readOnly			Allows or prevents user interaction
+ * @property renderOption	Custom render function for select options
+ * @property required			Whether or not a value is required
+ * @property useNatveSelect		Use the native <select> element if true
+ * @property value				The current value
+ * @property onBlur				Called when the input loses focus
+ * @property onChange			Called when the node's 'change' event is fired
+ * @property onClick			Called when the input is clicked
+ * @property onFocus			Called when the input is focused
+ * @property onKeyDown		Called on the input's keydown event
+ */
+export interface SelectProperties extends ThemeableProperties {
+	describedBy?: string;
+	disabled?: boolean;
+	formId?: string;
+	invalid?: boolean;
+	label?: string | LabelOptions;
+	multiple?: boolean;
+	name?: string;
+	options?: SelectOption[];
+	readOnly?: boolean;
+	renderOption?(option: SelectOption): DNode;
+	required?: boolean;
+	useNativeSelect?: boolean;
+	value?: string;
+	onBlur?(event: FocusEvent): void;
+	onChange?(option: SelectOption): void;
+	onClick?(event: MouseEvent): void;
+	onFocus?(event: FocusEvent): void;
+	onKeyDown?(event: KeyboardEvent): void;
+}
+
+// This should have a lookup somewhere
+const keys = {
+	escape: 27,
+	enter: 13,
+	space: 32,
+	up: 38,
+	down: 40,
+	home: 36,
+	end: 35
+};
+
+export const SelectBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export default class Select extends SelectBase<SelectProperties> {
+	private _focusedIndex = 0;
+	private _ignoreBlur = false;
+	private _open = false;
+	private _selectId: string = uuid();
+
+	private _onBlur (event: FocusEvent) { this.properties.onBlur && this.properties.onBlur(event); }
+	private _onClick (event: MouseEvent) { this.properties.onClick && this.properties.onClick(event); }
+	private _onFocus (event: FocusEvent) { this.properties.onFocus && this.properties.onFocus(event); }
+	private _onKeyDown (event: KeyboardEvent) { this.properties.onKeyDown && this.properties.onKeyDown(event); }
+
+	// native select events
+	private _onNativeChange (event: Event) {
+		const { options = [] } = this.properties;
+		const option = options.filter((option: SelectOption) => option.value === (<HTMLInputElement> event.target).value)[0];
```

Comment:
I'd break `value` out into its own variable so you don't have to dig into `event` every iteration of `filter`. I'd also use `find` instead of `filter` since you're always using the first item that satisfies the filter condition:

```ts
const value = (<HTMLInputElement> event.target).value;
const option = options.find((option: SelectOption) => option.value === value);
```